### ISSUE TITLE
VBET 0.8.1 - Size and Cost of Debug HUC 10s vs HUC 8s

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/riverscapes-tools#660]
It was interesting to see on these HUC 10s how manageable the sizes of projects were with and without debugging. For example the 1601020101 HUC 10 is 1.93 GB unzipped, but without the temp.zip is 0.99 GB. For me it poses a few questions (maybe for @MattReimer) before we get going:
1. Should we stick to plan of running VBET for HUC 8s or do for HUC10s instead? _I know this impacts all our tools, but with IGOs and potential to collate results across large areas (multiple projects or all projects) as well as new Warehouse and QRV improvements, I am less concerned. Plus, these might be slightly more manageable for users to download and work with. Just a question not a suggestion yet._
2. Should we commit to just storing the levlpath based VBETs as part of project or as their own projects? _I think we will want the individual levelpaths for QRiS more than full HUC VBETs and I think we will also want to use them in commercial grade vbet and/or Riverscape Studio eventually. Moreover, the largest intermediate rasters (e.g. `local_hand_levelpath.tif` or `vbet_evidence_levelpath.tif` are on the order 30 to 40 mb we might use to support slider tool (and many are 1 to 2 MB). This could make retrieving and serving just those rasters much more tractable?_ 
3. How much more is it costing us to write out (computation time and storage) the temp stuff? 

This is a non-urgent dialogue, but one we should hash out before we start rerunning VBET for entire 2/3 of USA.

**Comment by MattReimer** [comment]
1. From my point of view on a purely technical level it doesn't matter that much. They all run in the background. The question I'm most interested in is if we've solved the problem where the assembled HUC10 result is different from the HUC8 result (lots of tickets flying right now so sorry if I missed it). If that checks out then how we process and consume these things is more about user convenience than anything (like you said).

2. In the new system you will be able to pull only one layer from a project. Downloading the whole zip file is a relic of the old system. Don't know if that affects your answer.

3. The temp stuff should go away on production. It doubles (or more) the size of the project and includes a constly (and sometimes bug-prone) zipping step for potentially hundreds of thousands of files. Hardly anyone will ever need or want it. If there is something in there that people regularly need then we should consider writing it to another standard intermediate layer. Just my $0.02.

**Comment by joewheaton** [comment]
> 1. From my point of view on a purely technical level it doesn't matter that much. They all run in the background. The question I'm most interested in is if we've solved the problem where the assembled HUC10 result is different from the HUC8 result (lots of tickets flying right now so sorry if I missed it). If that checks out then how we process and consume these things is more about user convenience than anything (like you said).

Yes, sorry for lots of tickets. Your answer is reassuring. I will consult with @philipbaileynar. There is NOT an answer yet to whether or not HUC10 results are different then HUC8 results. Did you make a ticket for that somewhere @MattReimer? That is interesting. 

> 2. In the new system you will be able to pull only one layer from a project. Downloading the whole zip file is a relic of the old system. Don't know if that affects your answer.

Good to know... Not sure yet.

> 3. The temp stuff should go away on production. It doubles (or more) the size of the project and includes a constly (and sometimes bug-prone) zipping step for potentially hundreds of thousands of files. Hardly anyone will ever need or want it. If there is something in there that people regularly need then we should consider writing it to another standard intermediate layer. Just my $0.02.

Okay... helpful. Lets chat more later about this.  This is NOT a top priority at the moment.

**Comment by MattReimer** [comment]
> Did you make a ticket for that somewhere @MattReimer? That is interesting.

No, I confess I did not. All that kind of got sidetracked last week. Feel free to create one if you'd like. I'm not sure of the best wording. It's just a quick (ish) someone (shelby?) could do to see if we have any big problems. Maybe it's nothing but we won't know until we look.

**Comment by joewheaton** [comment]
Just capturing a conversation @philipbaileynar and @joewheaton had about this. We commented:

- We agree we need to understand if we get same answers at HUC 8 vs all of its HUC 10s being run for VBET. That is proceeding. The biggest unknown is how levelpaths that are not entirely contained within the HUC 10s perform.
- We discussed merits of running at HUC 10 vs HUC 8 and sort of left it at → lets just stick with HUC 8s.
- We also discussed merits of running HUC 10s AND keeping *ALL* levelpath stuff currently just in debugs. @philipbaileynar convinced me that we do not need to do this. Now that VBET is stabilizing and the composite HAND and VBET evidence look really good, we can take clipped versions of those and drive any QRiS slider games with. He also pointed out we should just bring functionality of HAND into VBET to support slider off of higher resolution data at desktop. 
- We need to convince ourselves that VBET actually runs okay still with debug mode turned off. 

@philipbaileynar we discussed a lot, but those seemed to be the main conclusions. 

In other words, in answer to my question number 1. at top of this ticket - No, we stick with HUC 8s. And in answer to question number 2. No, we keep the levelpaths as a debug mode thing only.

**Comment by MattReimer** [comment]
Sounds good. I'll close this ticket then. I think we covered all the bases and there are no more action items for this one.

**Comment by joewheaton** [comment]
Did you have any ticket for all  the HUC10s in a HUC8 compared to the HUC8 thing? If not, we should track that somewhere.

**Comment by MattReimer** [comment]
here: #663
